Mathlib/Analysis/NormedSpace/OperatorNorm/Prod.lean

```diff
@@ -0,0 +1,183 @@
+-/
+import Mathlib.Analysis.NormedSpace.OperatorNorm.Bilinear
+
+/-!
+# Operator norm: Cartesian products
+
+Interaction of operator norm with Cartesian products.
+
+-/
+
+suppress_compilation
+
+open Bornology
+open Filter hiding map_smul
+open scoped Classical NNReal Topology Uniformity
+
+-- the `ₗ` subscript variables are for special cases about linear (as opposed to semilinear) maps
+variable {𝕜 𝕜₂ 𝕜₃ E Eₗ F Fₗ G Gₗ 𝓕 : Type*}
+
+section SemiNormed
+
+open Metric ContinuousLinearMap
+
+variable [SeminormedAddCommGroup E] [SeminormedAddCommGroup Eₗ] [SeminormedAddCommGroup F]
+  [SeminormedAddCommGroup Fₗ] [SeminormedAddCommGroup G] [SeminormedAddCommGroup Gₗ]
+
+variable [NontriviallyNormedField 𝕜] [NontriviallyNormedField 𝕜₂] [NontriviallyNormedField 𝕜₃]
+  [NormedSpace 𝕜 E] [NormedSpace 𝕜 Eₗ] [NormedSpace 𝕜₂ F] [NormedSpace 𝕜 Fₗ] [NormedSpace 𝕜₃ G]
+  [NormedSpace 𝕜 Gₗ] {σ₁₂ : 𝕜 →+* 𝕜₂} {σ₂₃ : 𝕜₂ →+* 𝕜₃} {σ₁₃ : 𝕜 →+* 𝕜₃}
+  [RingHomCompTriple σ₁₂ σ₂₃ σ₁₃]
+
+variable [FunLike 𝓕 E F]
+
+namespace ContinuousLinearMap
+
+section OpNorm
+
+open Set Real
+
+@[simp]
+theorem opNorm_prod (f : E →L[𝕜] Fₗ) (g : E →L[𝕜] Gₗ) : ‖f.prod g‖ = ‖(f, g)‖ :=
+  le_antisymm
+      (opNorm_le_bound _ (norm_nonneg _) fun x => by
+        simpa only [prod_apply, Prod.norm_def, max_mul_of_nonneg, norm_nonneg] using
+          max_le_max (le_opNorm f x) (le_opNorm g x)) <|
+    max_le
+      (opNorm_le_bound _ (norm_nonneg _) fun x =>
+        (le_max_left _ _).trans ((f.prod g).le_opNorm x))
+      (opNorm_le_bound _ (norm_nonneg _) fun x =>
+        (le_max_right _ _).trans ((f.prod g).le_opNorm x))
+#align continuous_linear_map.op_norm_prod ContinuousLinearMap.opNorm_prod
+
+@[deprecated]
+alias op_norm_prod :=
+  opNorm_prod -- deprecated on 2024-02-02
+
+@[simp]
+theorem opNNNorm_prod (f : E →L[𝕜] Fₗ) (g : E →L[𝕜] Gₗ) : ‖f.prod g‖₊ = ‖(f, g)‖₊ :=
+  Subtype.ext <| opNorm_prod f g
+#align continuous_linear_map.op_nnnorm_prod ContinuousLinearMap.opNNNorm_prod
+
+@[deprecated]
+alias op_nnnorm_prod :=
+  opNNNorm_prod -- deprecated on 2024-02-02
+
+/-- `ContinuousLinearMap.prod` as a `LinearIsometryEquiv`. -/
+def prodₗᵢ (R : Type*) [Semiring R] [Module R Fₗ] [Module R Gₗ] [ContinuousConstSMul R Fₗ]
+    [ContinuousConstSMul R Gₗ] [SMulCommClass 𝕜 R Fₗ] [SMulCommClass 𝕜 R Gₗ] :
+    (E →L[𝕜] Fₗ) × (E →L[𝕜] Gₗ) ≃ₗᵢ[R] E →L[𝕜] Fₗ × Gₗ :=
+  ⟨prodₗ R, fun ⟨f, g⟩ => opNorm_prod f g⟩
+#align continuous_linear_map.prodₗᵢ ContinuousLinearMap.prodₗᵢ
+
+end OpNorm
+
+end ContinuousLinearMap
+
+namespace ContinuousLinearMap
+
+variable [RingHomIsometric σ₂₃] [RingHomIsometric σ₁₃]
+
+variable (E F G σ₁₃ σ₂₃)
+
+variable {E F G σ₁₃ σ₂₃}
+
+variable (𝕜 E Fₗ Gₗ)
+
+variable {𝕜 E Fₗ Gₗ}
+
+variable (F σ₁₂) [RingHomIsometric σ₁₂]
+
+variable {F σ₁₂}
+
+variable (𝕜 Fₗ)
+
+variable {𝕜 Fₗ}
+
+variable (σ₁₂ σ₂₃ E F G)
+
+set_option linter.uppercaseLean3 false
+
+variable {σ₁₂ σ₂₃ E F G}
+
+variable (𝕜 σ₁₂ σ₂₃ E Fₗ Gₗ)
+
+variable (Eₗ) {𝕜 E Fₗ Gₗ}
```

Comment:
Uhuh. I guess we can clean up after

Reply:
Oops, meant to take that out. What happened here is that since the original file has so many `variables` declarations overwriting each other in confusing ways, I did the split by repeatedly making copies and removing lemmas from each half, while keeping the section and variable declarations. (So this extract gives an idea of how bad the original file had got!)